esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6141,7 +6159,287 @@ bool CWsDfuEx::onDFUFileAccess(IEspContext &context, IEspDFUFileAccessRequest &r
             userDesc.setown(createUserDescriptor());
             userDesc->set(userID.str(), context.queryPassword(), context.querySignature());
         }
-        getFileAccess(context, userDesc, accessType, req, resp);
+        getFileAccess(context, userDesc, accessType, requestBase, resp.updateAccessInfo());
+    }
+    catch (IException *e)
+    {
+        FORWARDEXCEPTION(context, e,  ECLWATCH_INTERNAL_ERROR);
+    }
+    return true;
+}
+
+IGroup *CWsDfuEx::getDFUFileIGroup(const char *clusterName, ClusterType clusterType, const char *clusterTypeEx, StringArray &locations, StringBuffer &groupName)
+{
+    if (!locations.ordinality())
```

Comment:
I have a slight concern here.. 
If locations are provided.. and clusterName (e.g. "thor50") is provided, i.e. client is requesting to create a file on thor50 but is providing a part map..

Then the locations should at least be present in the "thor50" group.. if they're outside it, that's an error.
That's probably an edge case, but should be caught.
@wangkx -  please open a separate JIRA to address that.

Reply:
Also, what happens if the locations provided are an exact match for the cluster group?
It should use the cluster groupname.. it should construct a new one..

Reply:
https://track.hpccsystems.com/browse/HPCC-20697